### ISSUE TITLE
Remove duplicate hard-coded references to «var/generation» string

### ISSUE THREAD

**Issue by dmitry-fedyuk** [magento/magento2#2296]
Code duplication: \Magento\Framework\App\Filesystem\DirectoryList::getDefaultConfig() and \Magento\Framework\Console\Cli::doRun() use magic string «var/generation» instead of the constant \Magento\Framework\Code\Generator\Io::DEFAULT_DIRECTORY
- https://github.com/magento/magento2/blob/22629abe10f9c640fe47b6081d57fedccdea8e0f/lib/internal/Magento/Framework/Code/Generator/Io.php#L17
- https://github.com/magento/magento2/blob/22629abe10f9c640fe47b6081d57fedccdea8e0f/lib/internal/Magento/Framework/App/Filesystem/DirectoryList.php#L124
- https://github.com/magento/magento2/blob/22629abe10f9c640fe47b6081d57fedccdea8e0f/lib/internal/Magento/Framework/Console/Cli.php#L53

**Comment by mazhalai** [comment]
Thank you @dfediuk, we have created ticket MAGETWO-45198 to fix this.

**Comment by orlangur** [comment]
@dmitry-fedyuk thanks for your suggestion!

I wouldn't call identical string literals used a code duplication, especially in context of `lib/internal/Magento/Framework/Console/Cli.php` where it is only a part of error message. However, I've got a fancy idea out of your report: it would be nice to have a static test which forbids string literal usage if there is constant with such content. All those messy `TYPE_CODE = 'configurable'`, you know. Added to my TODO.

Nevertheless issue is fixed on `2.2.0` branch: https://github.com/magento/magento2/blob/2.2/lib/internal/Magento/Framework/App/Filesystem/DirectoryList.php#L152
